core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java

```diff
@@ -0,0 +1,163 @@
+package io.quarkus.deployment.ide;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
+import io.quarkus.runtime.util.JavaVersionUtil;
+
+public class IdeProcessor {
+
+    private static Map<String, List<Ide>> IDE_MARKER_FILES = new HashMap<>();
+    private static Map<Predicate<ProcessInfo>, Ide> IDE_PROCESSES = new HashMap<>();
+
+    static {
+        IDE_MARKER_FILES.put(".idea", Collections.singletonList(Ide.IDEA));
+        IDE_MARKER_FILES.put(".project", Arrays.asList(Ide.VSCODE, Ide.ECLIPSE));
+        IDE_MARKER_FILES = Collections.unmodifiableMap(IDE_MARKER_FILES);
+
+        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("idea")), Ide.IDEA);
+        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("code")), Ide.VSCODE);
+        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("eclipse")), Ide.ECLIPSE); // TODO: I have no idea if this is correct
+        IDE_PROCESSES = Collections.unmodifiableMap(IDE_PROCESSES);
+    }
+
+    @BuildStep
+    public EffectiveIdeBuildItem effectiveIde(IdeConfig ideConfig, IdeFileBuildItem ideFile,
+            IdeRunningProcessBuildItem ideRunningProcess) {
+        Ide result = null;
+        if (ideConfig.target == IdeConfig.Target.auto) {
+
+            // the idea here is to auto-detect the special files that IDEs create
+            // and also the running IDE process if need be
+
+            if (ideFile.getDetectedIDEs().size() == 1) {
```

Comment:
Good point, but it still doesn't help us determine Eclipse vs VSCode - which is why this whole dance is necessary.

Reply:
This detects runtime IDEs, but doesn't even copy their path, so it also doesn't look at executable modification date, which was my issue, because my IDE was not in my path. I figured this could be a further improvement.

Even then, if you decide to do this, watch out that often executables are links and I'm not sure their modification dates can be trusted without resolving the link.

Reply:
I am absolutely sure there will be improvements to be done in this area